pre_commit_hooks/check_fixed_library_versions.py

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env python
+
+import argparse
+
+from lxml import etree
+
+
+class PreCommitException(Exception):
+    pass
+
+
+def check_file(filename):
+    with open(filename, 'rb') as fd:
+        original_xml = fd.read()
+
+    xml_parser = etree.XMLParser(remove_blank_text=True)
+    parse_tree = etree.XML(original_xml, parser=xml_parser).getroottree()
+
+    added_libraries = set(el.attrib["Include"] for el in parse_tree.iter("{*}PlaceholderReference"))
+    fixed_version_libraries = set(el.attrib["Include"] for el in parse_tree.iter("{*}PlaceholderResolution"))
+
+    non_fixed_library_versions = added_libraries - fixed_version_libraries
```

Comment:
This misses the case where someone has explicitly set a library version as *, which makes it present in both sets but it is still not fixed. I'm not actually sure how this case happens but I found it in one of my projects while trying this out.

https://github.com/pcdshub/lcls-plc-lfe-motion/blob/23b9231c2051ec40c3b1df7d0ebdea9d20175611/plc-lfe-motion/lfe_motion/lfe_motion.plcproj#L186-L188

Reply:
I guess you can change this, although this is what a user would want in case they explicitly set a library to be the latest version.

Reply:
> Sorry for missing the PR- it's been a whirlwind here.

No problem!

Reply:
> this is what a user would want in case they explicitly set a library to be the latest version

Let's keep this exactly as-is then and I'll see if I want to make an optional check for some of my facility's repos